src/core/manager.c

```diff
@@ -3871,6 +3871,9 @@ static void manager_notify_finished(Manager *m) {
 
         bus_manager_send_finished(m, firmware_usec, loader_usec, kernel_usec, initrd_usec, userspace_usec, total_usec);
 
+        if (MANAGER_IS_SYSTEM(m) && detect_container() <= 0)
```

Comment:
the 2nd check really should be moved inside of watchdog_report_if_missing()

Reply:
A check in this form appears a few lines above… I considered moving this call up into that block, but it doesn't fit exactly. But moving half of this check here to a different place would introduce unnecessary difference from the rest of the logic in this function.